### ISSUE TITLE
Implement VM status check for each command which require VM interaction.

### ISSUE THREAD

**Issue by bhavin192** [minishift/minishift#1996]
### General information

  * Minishift version: `v1.12.0+c12e5117`
  * OS: Linux 
  * Hypervisor: KVM 

### Steps to reproduce

  1.  Make sure that `minishift` VM is created and is in `poweroff` state.
  2.  `$ minishift console`
 
### Expected
It should display error message like `minishift VM is not running`.

### Actual
Opens up browser window with IP address of dashboard.

I'm working on a patch for this.

**Comment by budhrg** [comment]
Thanks for opening the issue and working on it. Seems unusual behavior.

**Comment by amitkrout** [comment]
@bhavin192  I did ```minishift start``` on v1.13.1 and then ```minishift stop``` followed by ```minishift console```(mac os + firefox)
```
$ ./minishift console
Opening the OpenShift Web console in the default browser...
Cannot access the OpenShift console. Verify that Minishift is running. Error: Host is not running
$ echo $?
1
$ 
```
The message ```Opening the … ``` is only printed in the output, I don’t see the false dashboard open up in the browser. First console output line doesn’t look nice. IMO check on VM status should be done before executing this line ```Opening the OpenShift Web console in the default browser…```. 

If this is the case then the issue title should be something like ```VM status should be checked before popping up OpenShift Web console in browser```

**Comment by bhavin192** [comment]
@amitkrout tried same on my machine with v1.31.1, `start`, `stop`, `console` 
It still opening the browser window. 
I'm using `kvm` on Fedora 27

**Comment by LalatenduMohanty** [comment]
@bhavin192 It seems the behavior is different in macOS and in Fedora. I am not able to reproduce it on macOS.

```
$ minishift stop

$ minishift console
Opening the OpenShift Web console in the default browser...
Cannot access the OpenShift console. Verify that Minishift is running. Error: Host is not running

$ minishift status
Does Not Exist

$ minishift console
Opening the OpenShift Web console in the default browser...
Cannot access the OpenShift console. Verify that Minishift is running. Error: Machine 'minishift' does not exist. You need to run 'minishift start' first
```

**Comment by amitkrout** [comment]
I am able to reproduce it on Fedora 27. However ```minishift console``` doesn't pop up the false dashboard after deleting the stopped VM instance
```
$ ./minishift delete --force
Deleting the Minishift VM...
Minishift VM deleted.
[amit@localhost v1.13.1]$ ./minishift console
Opening the OpenShift Web console in the default browser...
Cannot access the OpenShift console. Verify that Minishift is running. Error: Machine 'minishift' does not exist. You need to run 'minishift start' first
[amit@localhost v1.13.1]$
```
Now checking the same on windows. Will update once done

**Comment by amitkrout** [comment]
Ran the same use case on ```win10+hyperv+v1.13.1```, however the vm start was too slow that it was still continuing after 5 mins (Might be resource issue, will check more usecae later). So i disabled it and executed the same on VirtualBox.
```
PS C:\v1.13.1> .\minishift.exe stop
Stopping local OpenShift cluster...
Cluster stopped.
PS C:\v1.13.1> .\minishift.exe console
Opening the OpenShift Web console in the default browser...
Cannot access the OpenShift console. Verify that Minishift is running. Error: Host is not running
PS C:\v1.13.1>
```
Looks fine. No false dashboard

**Comment by praveenkumar** [comment]
We need to add a basic check for the status of the machine before calling any other methods for commands which required to interact with VM. At this moment we are relying on around the driver plugin to do that check for us but each driver plugin is implemented differently using the driver interface and we can't be sure if every plugin behaves same way.

the following snip can be part of a function which can be used for each command as a check.

```golang
	s, err := d.GetState()
	if err != nil {
		return "", err
	}
	if s != state.Running {
		return "", drivers.ErrHostIsNotRunning
	}
```

**Comment by gbraad** [comment]
This is a duplicate of another issue. That issue contained a list of
commands that needed this check IIRC.

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by LalatenduMohanty** [comment]
+1 to abstract it as a single function.  I see this in 
```
$ git grep 'Running.String'
cmd/minishift/cmd/console.go:   if vmStatus != machineState.Running.String() {
cmd/minishift/cmd/status.go:    if vmStatus == state.Running.String() {
``` 
and we can improve the code in `minishift ip`.

**Comment by LalatenduMohanty** [comment]
We already have below functions in cmd/util/util.go
```
func IsHostRunning(driver drivers.Driver) bool {
	return drivers.MachineInState(driver, state.Running)()
}
```
and
```
func ExitIfNotRunning(driver drivers.Driver, machineName string) {
	running := IsHostRunning(driver)
	if !running {
		atexit.ExitWithMessage(0, fmt.Sprintf("Running this command requires a running '%s' VM, but no VM is running.", machineName))
	}
}
```
